### PR TITLE
args table integration

### PR DESCRIPTION
**Pull request by kroeder**

### CODE REVIEW

File: src/extract-args-types/extract-arg-types.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+interface ExtractedDirective {
+    inputs: Inputs;
+    outputs: Outputs;
+    propertiesWithoutDecorators: PropertiesWithoutDecorator;
+}
+
+type ExtractedDirectives = {
+    [key: string]: ExtractedDirective;
+};
```
Comment (kroeder) [inline src/extract-args-types/extract-arg-types.ts:20]: use type that is used in the webpack plugin

File: src/extract-args-types/extract-arg-types.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    };
+    table: {
+        category: "properties" | any;
+        defaultValue: { summary: string | undefined };
+        type: {
+            required: boolean;
+            summary: string | undefined;
+        };
+    };
+};
```
Comment (kroeder) [inline src/extract-args-types/extract-arg-types.ts:40]: check if the ArgsTable props are exported by @storybook/*

File: src/extract-args-types/extract-arg-types.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    description: prop.description,
+    defaultValue: prop.defaultValue,
+    type: { name: prop.type, required: true }, // todo get required
+    table: {
+        defaultValue: {
+            summary: prop.defaultValue,
+        },
+        type: {
+            summary: prop.type,
+            required: true, // todo get required
```
Comment (kroeder) [inline src/extract-args-types/extract-arg-types.ts:62]:
```suggestion
            required: false, // todo wait for isRequired field from the webpack plugin
```

File: src/extract-args-types/extract-arg-types.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    (window as any).STORYBOOK_ANGULAR_ARG_TYPES[name];
+
+const mapPropToArgsTableProp = (
+    prop: Property,
+    category: "inputs" | "outputs" | "properties"
+): ArgsTableProp => ({
+    name: prop.name,
+    description: prop.description,
+    defaultValue: prop.defaultValue,
+    type: { name: prop.type, required: true }, // todo get required
```
Comment (kroeder) [inline src/extract-args-types/extract-arg-types.ts:55]:
```suggestion
    type: { name: prop.type, required: false, /* todo wait for isRequired field from the webpack plugin */ },
```

File: src/extract-args-types/extract-arg-types.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        category,
+    },
+});
+
+// eslint-disable-next-line
+const mapPropsToArgsTableProps = (directive: ExtractedDirective): any => {
+    // const argsTableProps: ArgsTableProps = {};
+    const argsTableProps = [];
+
+    // eslint-disable-next-line ,no-restricted-syntax,no-restricted-syntax
```
Comment (kroeder) [inline src/extract-args-types/extract-arg-types.ts:73]: use outer for-loop with `for (['inputs', 'outputs', ...])`  to remove redundant code

File: src/test/angular-app/.storybook/preview.ts [hunk 6]
```diff
@@ -8,5 +10,9 @@ export const parameters = {
             date: /Date$/,
         },
     },
-    docs: { inlineStories: true },
+    docs: {
+        inlineStories: true,
+        extractArgTypes,
+        extractComponentDescription: undefined,
```
Comment (kroeder) [inline src/test/angular-app/.storybook/preview.ts:16]: figure out what this does

File: src/extract-args-types/extract-arg-types.ts (hunk 1 above)
Comment (kroeder) [reply src/extract-args-types/extract-arg-types.ts]: done

File: src/extract-args-types/extract-arg-types.ts (hunk 2 above)
Comment (kroeder) [reply src/extract-args-types/extract-arg-types.ts]: done

File: src/extract-args-types/extract-arg-types.ts (hunk 4 above)
Comment (kroeder) [reply src/extract-args-types/extract-arg-types.ts]: added comment in code

**Comment by kroeder** [comment]
I found types in `@storybook/components` and added them to this PR https://github.com/storybookjs/storybook/blob/f4b3a880e7f00bd1b28e7691d45bcc1c41b1cafe/lib/components/src/blocks/ArgsTable/types.ts

File: src/extract-args-types/extract-arg-types.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+});
+
+// eslint-disable-next-line
+const mapPropsToArgsTableProps = (
+    directive: ClassProperties
+): ExtendedArgType[] => {
+    const argsTableProps: ExtendedArgType[] = [];
+
+    for (const [categoryKey, category] of Object.entries(directive)) {
+        for (const [, property] of Object.entries<Property>(category)) {
```
Comment (Etienne-Buschong) [inline src/extract-args-types/extract-arg-types.ts:51]: `category` is actually of type `Property[]`, so you can just use `for (const property of category){...}`

**Pull request merged**